go/src/koding/kites/kloud/team/mongo.go

```diff
@@ -0,0 +1,132 @@
+package team
+
+import (
+	"fmt"
+	"strconv"
+
+	"koding/db/models"
+	"koding/db/mongodb/modelhelper"
+
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// adapter is a private interface used as an adapter for mongo singleton.
+type adapter interface {
+	GetAccount(username string) (*models.Account, error)          // AccountsColl
+	RelationshipCount(selector modelhelper.Selector) (int, error) // RelationshipColl
+	GetAllRelationships(selector modelhelper.Selector) ([]models.Relationship, error)
+	GetGroup(slugName string) (*models.Group, error)
+	GetGroupsByIds(ids ...bson.ObjectId) ([]*models.Group, error) // GroupsCollectionName
+}
+
+// MongoDatabase implements Database interface. This type is responsible for
+// communicating with Mongo database.
+type MongoDatabase struct {
+	adapter adapter
+}
+
+var _ Database = (*MongoDatabase)(nil)
+
+// NewMongoDatabase creates a new MongoDatabase instance.
+func NewMongoDatabase() *MongoDatabase {
+	return &MongoDatabase{
+		adapter: modelHelperAdapter{}, // use modelhelper package's singleton.
+	}
+}
+
+// Teams returns all teams stored in MongoDB database that matches a given
+// filter.
+func (m *MongoDatabase) Teams(f *Filter) ([]*Team, error) {
+	if f == nil {
+		f = &Filter{}
+	}
+
+	accountDB, err := m.adapter.GetAccount(f.Username)
+	if err != nil {
+		return nil, models.ResError(err, modelhelper.AccountsColl)
+	}
+
+	if f.Slug != "" {
+		return m.fetchOne(accountDB.Id, f.Username, f.Slug)
+	} else {
+		return m.fetchAll(accountDB.Id)
+	}
+}
+
+// fetchOne returns only specified team.
+func (m *MongoDatabase) fetchOne(accID bson.ObjectId, user, slug string) ([]*Team, error) {
+	groupDB, err := m.adapter.GetGroup(slug)
+	if err == mgo.ErrNotFound {
+		return []*Team{}, nil
+	} else if err != nil {
+		return nil, models.ResError(err, modelhelper.GroupsCollectionName)
+	}
+
+	belongs := modelhelper.Selector{
```

Comment:
most important reason i commented on this is, we dont have an index on these 3 fields. second but not that important one is, lets have these kind of logic in modelhelper